csaf-validation/src/main/kotlin/io/github/csaf/sbom/validation/tests/Tests.kt

```diff
@@ -260,6 +269,213 @@ object Test617MultipleScoresWithSameVersionPerProduct : Test {
     }
 }
 
+/**
+ * Implementation of
+ * [Test 6.1.14](https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#6114-sorted-revision-history).
+ */
+object Test6114SortedRevisionHistory : Test {
+    override fun test(doc: Csaf): ValidationResult {
+        // First, sort items ascending by number
+        val sortedByNumber =
+            doc.document.tracking.revision_history.sortedWith { h1, h2 ->
+                h1.number.compareVersionTo(h2.number)
+            }
+
+        // Then, check if it is sorted by date
+        val isSorted =
+            sortedByNumber.asSequence().zipWithNext { a, b -> a.date <= b.date }.all { it }
```

Comment:
Is `<=` sufficient or should it be strictly smaller? Note that the standard itself would first sort by date and then check that the numbers are also ordered correctly.

Reply:
The `<=` vs. `<` is only an issue if two releases would be emitted at the same time. I didn't see a test which explicitly checks for this. Is it forbidden or allowed? If it's allowed to have two releases with the same timestamp, this behavior is not defined precisely

Reply:
Similar to the issue below (but the other way around good vs. bad) @tschmidtb51 

The following test 6-1-14-12 is suppose to pass:
```
"tracking": {
      "current_release_date": "2021-07-22T10:00:00.000Z",
      "id": "OASIS_CSAF_TC-CSAF_2.0-2021-6-1-14-12",
      "initial_release_date": "2021-07-22T10:00:00.000Z",
      "revision_history": [
        {
          "date": "2021-07-22T10:00:00.000Z",
          "number": "2",
          "summary": "Second version."
        },
        {
          "date": "2021-07-22T10:00:00.000Z",
          "number": "1",
          "summary": "Initial version."
        }
      ],
      "status": "final",
      "version": "2"
    }
```

If we only  order by date, the order of the two are not determinable. It seems that in this case, it's also implicitly ordered by number if the date matches?

Reply:
Yes. We clarified that in [CSAF 2.1](https://github.com/oasis-tcs/csaf/blob/master/csaf_2.1/prose/share/csaf-v2.1-draft.md#sorted-revision-history). I'm not sure whether we already backported all tests related to that.

So: Order by date and then number; when comparing dates, take the timezone into account.

Reply:
Ah perfect, thanks for the quick reply. Implemented.